internal/build/lifecycle_execution.go

```diff
@@ -669,6 +683,8 @@ func (l *LifecycleExecution) Build(ctx context.Context, phaseFactory PhaseFactor
 		WithNetwork(l.opts.Network),
 		WithBinds(l.opts.Volumes...),
 		WithFlags(flags...),
+		If((!l.opts.Publish && l.hasExtensionsForBuild()), WithImage(l.opts.BuilderImage+"-extended")),
+		If((!l.opts.Publish && l.hasExtensionsForBuild()), WithUser(l.opts.Builder.UID(), l.opts.Builder.GID())),
```

Comment:
Aren't we removing the publish restriction?

Reply:
oops, forgot to change here, done